### ISSUE TITLE
Certificate not working (mTLS)

### ISSUE THREAD

**Issue by rogaha** [getinsomnia/insomnia#6946]
### Expected Behavior

SSL handshake working

### Actual Behavior

SSL: The Security framework only supports loading identities that are in PKCS#12 format.

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

8.5.1

### What operating system are you using?

macOS

### Operating System Version

14.2.1 (23C71)

### Installation method

download from insomnia.net

### Last Known Working Insomnia version

8.4.0

**Comment by jackkav** [comment]
We recently added support for macOS keychain certs, so its possible we caused a regression here. Can you please provide screen shots and as much info as possible in the issue for us to reproduce the issue.

**Comment by romkazor** [comment]
Confirm. 8.5.X broken. 8.4.5 working

**Comment by five510** [comment]
@jackkav 

Could you please confirm following information?

My working environment

```
Version: Insomnia 8.5.1
Build date: 12/21/2023
OS: Darwin arm64 23.1.0
Electron: 28.0.0
Node: 18.18.2
Node ABI: 119
V8: 12.0.267.8-electron.0
Architecture: arm64
```


The error `Error: Problem with the local SSL certificate` happened when using Client Certificates.  This feature actually worked in older version (8.4.x)

<img width="860" alt="スクリーンショット 2023-12-27 14 17 01" src="https://github.com/Kong/insomnia/assets/16430639/4e3b5e83-48ac-4ae4-b98c-a513a079adbd">

Following error is shown in Timeline when using crt + key files

<img width="719" alt="スクリーンショット 2023-12-27 14 17 29" src="https://github.com/Kong/insomnia/assets/16430639/47600e4c-02d7-4051-aeba-5eaa918aa273">

Following error is shown in Timeline when using p12 file.
<img width="847" alt="スクリーンショット 2023-12-27 14 16 45" src="https://github.com/Kong/insomnia/assets/16430639/8cbdf5f9-d31b-4b7c-a1d9-6498c50c9fc0">

The p12 file exists with the 644 permission.

**Comment by five510** [comment]
@jackkav 
Could you please confirm following information?

My working environment

```
Version: Insomnia 8.5.1
Build date: 12/21/2023
OS: Darwin arm64 23.1.0
Electron: 28.0.0
Node: 18.18.2
Node ABI: 119
V8: 12.0.267.8-electron.0
Architecture: arm64
```


The error `Error: Problem with the local SSL certificate` happened when using Client Certificates.  This feature actually worked in older version (8.4.x)

<img width="860" alt="スクリーンショット 2023-12-27 14 17 01" src="https://github.com/Kong/insomnia/assets/16430639/4e3b5e83-48ac-4ae4-b98c-a513a079adbd">

Following error is shown in Timeline when using crt + key files

<img width="645" alt="スクリーンショット 2023-12-27 14 35 13" src="https://github.com/Kong/insomnia/assets/16430639/82fda607-8fee-4902-be6b-6343a4b7a7c6">


Following error is shown in Timeline when using p12 file.
<img width="847" alt="スクリーンショット 2023-12-27 14 16 45" src="https://github.com/Kong/insomnia/assets/16430639/8cbdf5f9-d31b-4b7c-a1d9-6498c50c9fc0">

The p12 file exists with the 644 permission.

**Comment by jackkav** [comment]
@notjaywu any ideas?

**Comment by notjaywu** [comment]
@jackkav Yes, it is regressioin.

1. SecureTransport on macOS only supports loading identities that are in PKCS#12 format. It fails to read the cert and key pair.
2. For the `.p12` issue, the `.p12` file should be added to the system keychain manually, as the error message said, it is required by SecureTransport.

**Comment by szechyjs** [comment]
I can confirm client certs are busted in the latest version for me on macOS.

I receive the following error `Error: Failure when receiving data from the peer`

```
* Preparing request to https://foo.azurewebsites.net/v1/foo
* Current time is 2024-01-04T21:41:33.253Z
* Enable automatic URL encoding
* Adding SSL P12 certificate
* Using HTTP/2
* Enable SSL validation
* Enable cookie sending with jar of 3 cookies
* Too old connection (230 seconds), disconnect it
* Connection 7 seems to be dead!
* Closing connection 7
* Connected to foo.azurewebsites.net (40.70.27.35) port 443 (#11)
* ALPN, offering h2
* ALPN, offering http/1.1
* Client certificate: xxxx
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: *.azurewebsites.net
* Server certificate: Microsoft Azure TLS Issuing CA 01
* Server certificate: DigiCert Global Root G2

> GET /v1/foo HTTP/1.1
> Host: foo.azurewebsites.net
> User-Agent: insomnia/8.5.1
> Accept: application/json
> Authorization: Bearer xxx

* SSLRead() return error -9800
* Closing connection 11
```

```
Version: Insomnia 8.5.1
Build date: 12/21/2023
OS: Darwin x64 22.6.0
Electron: 28.0.0
Node: 18.18.2
Node ABI: 119
V8: 12.0.267.8-electron.0
Architecture: x64
```

**Comment by pinkyjpainadath** [comment]
I am facing the same issue. is there any fix or workaround available for this

**Comment by DennyLoko** [comment]
@pinkyjpainadath 
> I am facing the same issue. is there any fix or workaround available for this

In the meantime, I've downgraded to version 8.4.5 until the fix is ready.

**Comment by pinkyjpainadath** [comment]
@DennyLoko  how did you downgrade. do we need to uninstall the current one and download the old package from GIT ?

**Comment by markosamuli** [comment]
Not sure if I deleted the app first, but downloaded the [8.4.5 release](https://github.com/Kong/insomnia/releases/tag/core%408.4.5) from GitHub that fixed the issue for me.

Also disabled the automated updates in settings:

![CleanShot 2024-01-09 at 09 09 39@2x](https://github.com/Kong/insomnia/assets/61325/be152a29-2187-41ee-bf6e-4540e325a79e)

**Comment by notjaywu** [comment]
[Insomnia 8.5.2-beta.0 📦](https://github.com/Kong/insomnia/releases/tag/core%408.5.2-beta.0) is out to fix this issue, please take a try, thank you!

**Comment by sathyarajagopal** [comment]
@notjaywu In 8.5.2-beta.0, I am able to successfully make HTTPS calls with client cert (.p12). Thank you for fixing this.

**Comment by notjaywu** [comment]
It has now been fixed in 8.6.0+.
